0022-lnd-plan-of-attack.adoc

```diff
@@ -0,0 +1,448 @@
+From an API PoV, we can therefore assume that there is no accept/decline stage with user interaction.
+
+==== Lightning is not a public ledger
+
+This problem is mostly relevant from an integration-design point of view.
+Our current implementation has a module called `btsieve` which takes care of fetching blocks and scanning them for certain `TransactionPattern`s.
+Lightning doesn't have any blocks, hence the integration with Lightning needs to take different form here.
+
+==== Standard LN does not support payments without the other party knowing the pre-image
+
+In a standard LN setup, parties can only receive payments if their node is knowledgable of the pre-image relevant in this payment.
+In an atomic swap scenario, this would only be given if the Lightning is the beta-ledger.
+In such scenario, Bob would send a payment on Lightning and Alice can immediately take it because she knows the secret.
+
+==== Our current REST API endpoints are not capable of expressing the concept of a protocol per ledger
+
+This is again an integration-design problem.
+We need to define an approach for how to introduce lightning support without unnecessary breaking API-changes.
+
+== Research
+
+=== Addressing above mentioned problems
+
+==== Re-use "half the protocol"
+
+This was effectively solved by the naming protocol spike.
+We can refer to the protocols of each ledger separately.
+This allows us to define a HALight protocol without having to worry about the other side.
+Valid combinations shall be determined by implementations.
+
+==== Messaging protocol
+
+This problem will be solved by a different spike.
+For this current one, we will make the following assumptions (which will influence the design of the new messaging):
+
+- Adding support for the HALight protocol can be done by _adding_ libp2p protocols instead of modifying existing ones (as it is currently the case)
+
+This should allow us to finish this spike and once the new communication protocol is defined, add support for HALight to it based on the needs that will be revealed in this spike.
+
+==== Lightning is not a public ledger
+
+At the moment, the implementation of the rfc003 protocol is based on the `HtlcEvents` trait.
+Said trait exposes the events that can happen as part of the protocol execution.
+
+The implementations of that make use of the `btsieve` module which cannot be used for Lightning as already mentioned above.
+Hence, the integration of Lightning will likely require adding an "Events" traits, conceptually similar to `HtlcEvents`.
+`HtlcEvents` is already specific to a single ledger, so we can simply add a trait `halight::Events` that describes all the events that can happen during the execution of the HALight protocol.
+
+Note: `HtlcEvents` will be decomposed into single function traits with https://github.com/comit-network/comit-rs/issues/1779[#1779].
```

Comment:
I feel that with the introduction of HALight then the name `HtlcEvents` for HAN does not fit anymore.
Even with this new issue, I think the terminology `Htlc` is going to be confusing.
Maybe similarly to what you propose for HALight, we should update comit-network/comit-rs#1779 to create new traits which are `han::Funded` instead of `HtlcFunded`.

Reply:
It really depends on which issue happens first.
I'd expect everyone to follow along here good enough to get this :)